bson/bsoncodec/default_value_encoders.go

```diff
@@ -713,3 +746,12 @@ func (dve DefaultValueEncoders) CodeWithScopeEncodeValue(ec EncodeContext, vw bs
 	}
 	return dw.WriteDocumentEnd()
 }
+
+// IsImplementationNil returns if val is a nil pointer and inter is implemented on a concrete type
+func IsImplementationNil(val reflect.Value, inter reflect.Type) bool {
```

Comment:
I'd prefer to not have this be part of our public API. If needed, we can copy it in the mgocompat package.

Reply:
done